### PR TITLE
hmc7044: Extend interface

### PR DESCRIPTION
**Pull request by cristiantpop**
Extend interface, with equivalent functions, found in linux driver.
They will be needed in the JESD autoconfiguration project.

Signed-off-by: Cristian Pop <cristian.pop@analog.com>

### CODE REVIEW

File: drivers/frequency/hmc7044/hmc7044.c [hunk 1]
```diff
@@ -290,6 +290,64 @@ uint32_t hmc7044_calc_out_div(uint32_t rate,
...
+{
+	uint32_t div;
+
+	if (chan >= dev->num_channels)
+		return FAILURE;
+
+	div = hmc7044_calc_out_div(rate, dev->pll2_freq);
+	dev->channels[chan].divider = div;
+
+	hmc7044_write(dev, HMC7044_REG_CH_OUT_CRTL_1(chan),
```
Comment (mirceacaprioru) [inline drivers/frequency/hmc7044/hmc7044.c]: can we check that the writes succeded and return that ?

ret = hmc7044_write

return ret ?

**Comment by cristiantpop** [comment]
V2:

-  check "hmc7044_write" return value.

File: drivers/frequency/hmc7044/hmc7044.c [hunk 2]
```diff
@@ -290,6 +290,70 @@ uint32_t hmc7044_calc_out_div(uint32_t rate,
...
+
+	div = hmc7044_calc_out_div(rate, dev->pll2_freq);
+	dev->channels[chan].divider = div;
+
+	ret = hmc7044_write(dev, HMC7044_REG_CH_OUT_CRTL_1(chan),
+			    HMC7044_DIV_LSB(div));
+	if(ret < 0)
+		return ret;
+
+	ret = hmc7044_write(dev, HMC7044_REG_CH_OUT_CRTL_2(chan),
```
Comment (mirceacaprioru) [inline drivers/frequency/hmc7044/hmc7044.c]: return directly here ? 

return hmc7044_write ...

**Comment by cristiantpop** [comment]
V3:

- return directly "hmc7044_write()" value, for "hmc7044_clk_set_rate()" function. In this way, an if statement is removed.

**Comment by mirceacaprioru** [comment]
will merge later today if there are no other comments.

**Comment by cristiantpop** [comment]
V4:
- Get changes from master into current branch, to remove Travis errors

**Pull request merged**